src/services/fetch.rs

```diff
@@ -27,12 +46,48 @@ pub enum Credentials {
     SameOrigin,
 }
 
+/// Type to set mode for fetch.
+#[derive(Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum Mode {
+    /// `same-origin` value of mode.
+    SameOrigin,
+    /// `no-cors` value of mode.
+    NoCors,
+    /// `cors` value of mode.
+    Cors,
+    /// `navigate` value of mode.
+    Navigate,
```

Comment:
Looks like `navigate` is not a supported mode: https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch

Reply:
done!